src/riscv/lib/src/machine_state/block_cache/block/interpreted.rs

```diff
@@ -136,6 +136,7 @@ impl<MC: MemoryConfig, M: ManagerBase> Block<MC, M> for Interpreted<MC, M> {
         &mut self,
         core: &mut MachineCoreState<MC, M>,
         mut instr_pc: Address,
+        _max_steps: u64,
```

Comment:
You need to pass this along to `run_block_inner`, I think. Otherwise, this function may run more steps than passed in.